core/src/main/java/hudson/model/AbstractProject.java

```diff
@@ -1207,7 +1207,13 @@ public boolean checkout(AbstractBuild build, Launcher launcher, BuildListener li
             return true;    // no SCM
 
         FilePath workspace = build.getWorkspace();
-        workspace.mkdirs();
+        if(workspace!=null){
+            workspace.mkdirs();
+        }
+        else{
+            return true;
```

Comment:
Sounds like an error suppression. Could you please explain this change a bit?

Reply:
Don't think the error will be suppressed. Should be caught in AbstracBuild during onCheckout. Here just avoiding a NPE for null workspace. Or if dont want to suppress it, we can throw a runtime excpetion with proper message if you suggest.